scripts/print_dev_env.sh

```diff
@@ -0,0 +1,96 @@
+#!/bin/sh
+#
+# print_dev_env.sh
+#
+# This file is part of mbed TLS (https://tls.mbed.org)
+#
+# Copyright (c) 2014-2016, ARM Limited, All Rights Reserved
+#
+# Purpose
+#
+# To print out all the relevant information about the development encironment.
+#
+# This includes:
+#   - architecture of the system
+#   - type and version of the operationg system
+#   - version of armcc, gcc-arm and gcc compilers
+#   - version of libc, clang, asan and valgrind
+#   - version of gnuTLS and OpenSSL
+
+echo "\n1) Operating system and architecture:"
+uname -a
+
+if [ `which armcc` > /dev/null ]; then
```

Comment:
Avoid using "which" to check if an executable exists, it is an external command and its effects can differ on other systems, take a look at this link http://stackoverflow.com/questions/592620/check-if-a-program-exists-from-a-bash-script. I would suggest the use of built-in  "hash".